consensus/misc/eip1559/eip1559.go

```diff
@@ -57,7 +57,14 @@ func VerifyEIP1559Header(config *params.ChainConfig, parent, header *types.Heade
 
 // CalcBaseFee calculates the basefee of the header.
 // The time belongs to the new block to check if Canyon is activted or not
-func CalcBaseFee(config *params.ChainConfig, parent *types.Header, time uint64) *big.Int {
+func CalcBaseFee(config *params.ChainConfig, parent *types.Header, time uint64) (response *big.Int) {
+	defer func() {
+		// If the base fee response is below the floor, intercept the return and return the floor instead.
+		if config.Celo != nil {
+			response = math.BigMax(response, new(big.Int).SetUint64(config.Celo.EIP1559BaseFeeFloor))
+		}
+	}()
```

Comment:
Using a defer is a creative way to handle all the different returns without refactoring the function. A bit unintuitive but clever!

Reply:
I agree that is not intuitive. I will add another comment more as a warning for the return, just in case